app/models/pool.rb

```diff
@@ -0,0 +1,60 @@
+class Pool < ApplicationRecord
+  belongs_to :event
+
+  def self.recalculate(event)
+    # When deleting an event, its ID is not in "this_and_following_events".
+    Pool.where(event_id: event.id).delete_all if event.destroyed?
+
+    later_event_ids = event.this_and_following_events.pluck(:id)
```

Comment:
This way of finding the pool items to delete leads to an issue.

Let's say you have an event each month of the year.
You add a 13 event by mistake in Jan
You move it to Dec.

Then this code will look for all the events following this event after we updated its date to dec.
But we actually need to recalculate the pool for all events that were following the event before we changed its date (so Jan - Dec and not Dec only)

Reply:
Good point. We need to find the earliest date from before and after the change.

Reply:
Maybe move the delete pool associated to the event into a Rails `dependent` scenario and make `recalculate` accept something that can go into a `sort_column` I mean `.where("sort_column >= ?", sort_column)`